substrate/frame/tips/src/lib.rs

```diff
@@ -611,4 +630,38 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 			Tips::<T, I>::insert(hash, new_tip)
 		}
 	}
+
+	/// Ensure the correctness of the state of this pallet.
+	///
+	/// This should be valid before and after each state transition of this pallet.
+	///
+	/// ## Invariants:
+	/// 1. Reasons exists for each Tip[`OpenTip.reason`], this implies equal lenth of entries in storage.
+	/// 2. If `OpenTip.finders_fee` is true, then OpenTip.deposit should be greater that zero.
+	#[cfg(any(feature = "try-runtime", test))]
+	pub fn do_try_state() -> Result<(), TryRuntimeError> {
+		let reasons = Reasons::<T, I>::iter_keys().collect::<Vec<_>>();
+		let tips = Tips::<T, I>::iter_keys().collect::<Vec<_>>();
+
+		ensure!(reasons.len() == tips.len(), TryRuntimeError::Other("Equal length of entries in 'Tips' and 'Reasons` Storage"));
```

Comment:
Maybe try to use only one set of ticks :)
```suggestion
		ensure!(reasons.len() == tips.len(), TryRuntimeError::Other("Equal length of entries in `Tips` and `Reasons` Storage"));
```